apiserver/facades/client/modelgeneration/modelgeneration.go

```diff
@@ -126,6 +147,13 @@ func (api *API) TrackBranch(arg params.BranchTrackArg) (params.ErrorResults, err
 	if !isModelAdmin && !api.isControllerAdmin {
 		return params.ErrorResults{}, common.ErrPerm
 	}
+	// Ensure we guard against the numUnits being greater than 0 and the number
+	// units/applications greater than 1. This is because we don't know how to
+	// topographically distribute between all the applications and units,
+	// especially if an error occurs whilst assigning the units.
+	if arg.NumUnits > 0 && len(arg.Entities) > 1 {
+		return params.ErrorResults{}, errors.Errorf("num of units allowed when specifying multiple entities")
```

Comment:
I would make this explicit. maybe "number of units and unit IDs can not be specified at the same time".

Reply:
Righto, sounds like a better change tbh